framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -986,7 +986,9 @@ protected void checkPurity(MethodTree node) {
       reportPurityErrors(r, node, kinds);
     }
 
-    if (suggestPureMethods) {
+    // This may make a suggestion about an implicit default constructor, which is not considered
+    // synthetic.
+    if (suggestPureMethods && !TreeUtils.isSynthetic(node)) {
```

Comment:
Should the documentation of `isSynthetic` be update?

The implementation looks like it handles implicit default constructors.  But maybe a "generated" constructor is not an implicit default constructor?

Reply:
You are right: `Flags.GENERATEDCONSTR` is exactly for the default constructor.